dapp/src/components/buySell/ApproveSwap.js

```diff
@@ -41,14 +41,7 @@ const ApproveSwap = ({
     ContractStore.update((s) => {
       s.approvalNeeded = !approvalNeeded
     })
-  }, [
-    selectedSwap,
-    formHasErrors,
```

Comment:
other vars don't affect the contents in the body of the function... removing them